src/main/java/org/spongepowered/api/command/AbstractCommandModule.java

```diff
@@ -0,0 +1,366 @@
+ *
+ * <p>A command module is <b>not</b> re-usable.</p>
+ */
+public abstract class AbstractCommandModule {
+
+    private Binder binder = new SimpleBinder();
+    @SuppressWarnings("NullableProblems")
+    private PluginContainer plugin;
+    @SuppressWarnings("NullableProblems")
+    private CommandManager manager;
+    private InputTokenizer inputTokenizer = InputTokenizer.quotedStrings(false);
+    private TextSerializer textSerializer = TextSerializers.FORMATTING_CODE;
+    @Nullable private CommandThrowableProcessor throwableProcessor;
+    @Nullable private TextFunction textFunction;
+    private CommandAuthorizer authorizer = CommandAuthorizer.CALLABLE;
+    private boolean configured;
+    private boolean registered;
+
+    protected AbstractCommandModule() {
+    }
+
+    /**
+     * Contributes bindings and other configurations for this module to {@code binder}.
+     *
+     * @param plugin The plugin
+     * @param manager The command manager
+     */
+    @SuppressWarnings("ConstantConditions")
+    public final void configure(PluginContainer plugin, CommandManager manager) {
+        checkState(this.manager == null, "Re-entry is not allowed.");
+        this.plugin = checkNotNull(plugin, "plugin");
+        this.manager = checkNotNull(manager, "manager");
+        this.preConfigure();
+        this.configure();
+        // Create a delegating read-only binder
+        this.binder = new ReadOnlyDelegateBinder(this.binder);
+        this.configured = true;
+    }
+
+    private void preConfigure() {
+        // Register a few of the most common arguments by default
+        this.bind(CommandSource.class).toProvider((source, args, modifiers) -> source);
+        this.bind(CommandArgs.class).toProvider((source, args, modifiers) -> args);
+        this.bind(Game.class).toProvider((source, args, modifiers) -> Sponge.getGame());
+        this.bind(Logger.class).toProvider((source, args, modifiers) -> this.plugin.getLogger());
+    }
+
+    private void noChangingRegistration(String what) {
+        checkState(!this.registered, "The %s cannot be changed after commands have been registered", what);
+    }
+
+    /**
+     * Determines if this command module can be installed or configured
+     * through a {@link CommandManager}.
+     *
+     * @param install {@code true} if we should check if this module
+     *     can be installed
+     * @return {@code true} if this command module can be installed
+     *     or configured
+     */
+    @SuppressWarnings("ConstantConditions")
+    public final boolean canBeInstalledOrConfigured(boolean install) {
+        return install ? (this.plugin == null && this.manager == null && !this.configured) : !this.configured;
+    }
+
+    /**
+     * Configure this command module.
+     *
+     * <p>You <b>must</b> {@link #register(Object) register commands} <i>after</i>
+     * you have configured any bindings, input tokenizer, etc.</p>
+     */
+    @SuppressWarnings("WeakerAccess")
```

Comment:
Ehhh you could probably just disable this inspection in your IDE instead. If we suppressed this warning consistently these would be plastered all over Sponge.